[SPARK-13382][DOCS][PYSPARK] Update pyspark testing notes in build docs

holdenk opened the pull request:
## What changes were proposed in this pull request?

The current build documents don't specify that for PySpark tests we need to include Hive in the assembly otherwise the ORC tests fail.
## How was the this patch tested?

Manually built the docs locally. Ran the provided build command follow by the PySpark SQL tests.

![pyspark2](https://cloud.githubusercontent.com/assets/59893/13190008/8829cde4-d70f-11e5-8ff5-a88b7894d2ad.png)

holdenk committed:
Add a quick note about needing to build with hive support for PySpark's full tests

holdenk committed:
Make it a sub section instead

holdenk committed:
Commas

SparkQA commented:
**[Test build #51577 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/51577/consoleFull)** for PR 11278 at commit [`59afdb6`](https://github.com/apache/spark/commit/59afdb65b573893097fbbbf0e16a601a530bd20a).

SparkQA commented:
**[Test build #51577 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/51577/consoleFull)** for PR 11278 at commit [`59afdb6`](https://github.com/apache/spark/commit/59afdb65b573893097fbbbf0e16a601a530bd20a).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/51577/
Test PASSed.

holdenk committed:
Merge branch 'master' into SPARK-13382-update-pyspark-testing-notes-r2

SparkQA commented:
**[Test build #52586 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/52586/consoleFull)** for PR 11278 at commit [`36d1397`](https://github.com/apache/spark/commit/36d13972625f3c73b58996feea7e2d33881d4398).

SparkQA commented:
**[Test build #52586 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/52586/consoleFull)** for PR 11278 at commit [`36d1397`](https://github.com/apache/spark/commit/36d13972625f3c73b58996feea7e2d33881d4398).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/52586/
Test PASSed.

holdenk commented:
Maybe @jkbradley could merge this? Super simple update.

JoshRosen commented:
This is going to be outdated soon, since there's ongoing effort to remove the need to build assemblies before running tests.

holdenk commented:
Ok - I can close this if we think thats going to be fixed soon (e.g. before the next release).

holdenk commented:
So looking at the PR which went in - it seems we don't need the assembly jar but we do need to explicitly build with package & hive support for the PySpark tests which isn't necessarily clear in the current docs so I think this remains a reasonable change - what are your thoughts @JoshRosen ?

holdenk committed:
Merge branch 'master' into SPARK-13382-update-pyspark-testing-notes-r2

SparkQA commented:
**[Test build #55746 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55746/consoleFull)** for PR 11278 at commit [`0853b07`](https://github.com/apache/spark/commit/0853b079ddd7e86830a0504abf0eb2dc4ef214b2).

SparkQA commented:
**[Test build #55746 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55746/consoleFull)** for PR 11278 at commit [`0853b07`](https://github.com/apache/spark/commit/0853b079ddd7e86830a0504abf0eb2dc4ef214b2).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/55746/
Test PASSed.

holdenk commented:
re-ping @JoshRosen thoughts?

holdenk commented:
cc @markgrover who updated the building docs most recently, what are your thoughts?

markgrover commented on docs/building-spark.md:
```diff
@@ -123,6 +123,21 @@ To produce a Spark package compiled with Scala 2.10, use the `-Dscala-2.10` prop
 
+# PySpark Tests with Maven
+
+If you are building PySpark and wish to run the PySpark tests you will need to build an assembly JAR and also ensure you have built with hive support.
```
Yeah, I agree with @JoshRosen since assembly jars are now gone. I think the hive support part still applies but we should take out the part to build an assembly jar.

The ```hadoop-2.4``` profile sets the ```hadoop.version``` so I don't think we need to specify the latter explicitly. I am also not sure, at the top of my head, if we need the ```hive-thriftserver``` profile for running python tests so it may make sense to see if just specifying the hive profile would suffice.

markgrover commented on docs/building-spark.md:
```diff
@@ -123,6 +123,21 @@ To produce a Spark package compiled with Scala 2.10, use the `-Dscala-2.10` prop
+
+    ./python/run-tests --python-executables=python --modules=pyspark-sql
+
+**Note:** You can also run Python tests with an sbt build, provided your assembly JAR includes hive support.
```
Again, we should remove the reference to assembly jar from here, something like 'provided you build spark with hive support' should suffice.

markgrover commented:
Left some comments, overall, it's a change in the right direction. Thanks for working on this, @holdenk!

holdenk committed:
Update remove assembly and simplify build instructions

holdenk committed:
Merge branch 'master' into SPARK-13382-update-pyspark-testing-notes-r2

holdenk committed:
simplify

SparkQA commented:
**[Test build #56268 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56268/consoleFull)** for PR 11278 at commit [`47045c2`](https://github.com/apache/spark/commit/47045c2353206f590d68d131df4dd2f14d1e6125).

SparkQA commented:
**[Test build #56268 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/56268/consoleFull)** for PR 11278 at commit [`47045c2`](https://github.com/apache/spark/commit/47045c2353206f590d68d131df4dd2f14d1e6125).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/56268/
Test PASSed.

holdenk commented:
Rad, wanted to make sure people were still up for adding a note about the hive requirement even though we got rid of the assembly requirement. I've changed it to just say built with hive and dropped the extra bits from the build argument (were there to match the rest of the build reference in the doc but for Python tests probably not relevant).

markgrover commented:
LGTM. +1 (non-binding)

holdenk commented:
Since we've branched for 2.0 would this maybe be a good time to merge this?

holdenk commented:
ping @jkbradley

holdenk commented:
ping @JoshRosen ?

holdenk commented:
If no one objects maybe @markgrover could just go ahead and merge this?

markgrover commented:
Would love to help, but I am not a committer:-)

vanzin commented:
Merging to master / 2.0.
